webview/src/shared/components/messagesMenu/MessagesMenuOption.tsx

```diff
@@ -1,24 +1,38 @@
 import React from 'react'
+import cx from 'classnames'
 import { MessageFromWebview } from 'dvc/src/webview/contract'
 import { VSCodeDivider } from '@vscode/webview-ui-toolkit/react'
 import styles from './styles.module.scss'
 import { sendMessage } from '../../vscode'
 
 export interface MessagesMenuOptionProps {
   id: string
+  hideOnClick?: () => void
   label: string
   message?: MessageFromWebview
-  hidden?: boolean
+  disabled?: boolean
   divider?: boolean
   keyboardShortcut?: string
 }
 
 export const MessagesMenuOption: React.FC<
   MessagesMenuOptionProps & { onOptionSelected?: () => void }
-> = ({ label, message, divider, onOptionSelected, keyboardShortcut }) => {
+> = ({
+  label,
+  message,
+  hideOnClick,
+  disabled,
+  divider,
+  onOptionSelected,
+  keyboardShortcut
+}) => {
   const sendTheMessage = () => {
+    if (disabled) {
```

Comment:
[F] This is why we need control of `hideOnClick`. If the behaviour is wrapped up in `positionContextMenuAndDisableEvents` then we cannot stop the context menu from closing if a disabled item is click (not the normal behaviour)